### PR TITLE
Enhancement: Support JSON API content type

### PR DESCRIPTION
**Pull request by khornberg**

### CODE REVIEW

**Comment by khornberg** [comment]
@mattjmorrison-imt Not sure why the default is not being reported. This is ready otherwise.

**Comment by Mr-TaylorHobbs** [comment]
@mattjmorrison-imt The default will not report... Could we get some assistance with this?

**Comment by Mr-TaylorHobbs** [comment]
@mattjmorrison-imt We can't slap labels on this PR, but it is `ready_for_review`

File: generic_request_signer/factory.py [hunk 1]
```diff
@@ -88,7 +89,10 @@ def _get_data_payload(self, request_headers):
         if self.raw_data and not self.method_uses_querystring():
             content_type = request_headers.get("Content-Type")
             encoding_func = self.content_type_encodings.get(content_type, default_encoding)
-            return encoding_func(self.raw_data)
+            encoded_data = encoding_func(self.raw_data)
+            if not six.PY2 and isinstance(self.raw_data, str):
+                return encoded_data.encode()
```
Comment (mattjmorrison-imt) [inline generic_request_signer/factory.py]: This change makes me a little bit nervous about negatively impacting other things running Python3... so 2 questions:

1. Why is this needed?
2. Why have we not needed this change until now?

Comment (Mr-TaylorHobbs) [inline generic_request_signer/factory.py]: The specific error that we get is `POST data must be bytes, not str`. We ran into this issue when we were trying to run the claims client with Python 3. I do not know how we have avoided this issue so far, but it looked like most other things were a `dict` when they reached that part of the code.
